packages/server/validators/events.js

```diff
@@ -0,0 +1,203 @@
+    .toDate()
+    .withMessage("End date is not in a valid date format")
+    .custom(async (endDate, { req }) => {
+      if (req.body.startDate) {
+        if (endDate < req.body.startDate) {
+          throw new Error("End date is before new start date");
+        }
+      }
+    }),
+];
+
+const updateEventValidator = [
+  //Title checks
+  body("title", "Invalid title")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ min: 1, max: 255 })
+    .withMessage("Title length must be between 1 to 255 characters"),
+  //Description checks
+  body("description", "Invalid description")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ max: 2000 })
+    .withMessage("Description max length is 2000 characters"),
+  //Location checks
+  body("location", "Invalid location")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ max: 255 })
+    .withMessage("Location max length is 255 characters"),
+  //Thumbnail checks
+  body("thumbnail", "Invalid thumbnail")
+    .trim()
+    .escape()
+    .optional()
+    .isLength({ max: 255 })
+    .withMessage("Thumbnail max length is 255 characters")
+    .custom(async (value) => {
+      if (!/\.(jpg|png|JPG|PNG|JPEG|jpeg)$/.test(value)) {
+        throw new Error("Thumbnail is not a valid image file");
+      }
+    }),
+  //Start Date checks
+  //Convert end date to a Date object for easier comparison
+  check("endDate").optional().toDate(),
+  body("startDate", "Invalid dates")
+    .trim()
+    .escape()
+    .optional()
+    .isISO8601()
+    .toDate()
+    .withMessage("Start date is not in a valid date format")
+    .custom(async (startDate, { req }) => {
+      //If new end date is provided, check if start date is before the end date
+      if (req.body.endDate) {
+        if (startDate > req.body.endDate) {
+          throw new Error("Start date is after new end date");
+        }
+      } else {
+        //Checks if new start date is before kept end date
+        const currEvent = await EventController.getEvent(req.params.eventId);
+        if (currEvent.endDate != null && startDate > currEvent.endDate) {
+          throw new Error("Start date is after end date");
+        }
+      }
+    }),
+  //End Date checks
+  //Similar checks to start sate but makes sure end date is after the start
+  check("startDate").optional().toDate(),
+  body("endDate", "Invalid end date")
+    .trim()
+    .escape()
+    .optional()
+    .isISO8601()
+    .toDate()
+    .withMessage("End date is not in a valid date format")
+    .custom(async (endDate, { req }) => {
+      if (req.body.startDate) {
+        if (endDate < req.body.startDate) {
+          throw new Error("End date is before new start date");
+        }
+      } else {
+        const currEvent = await EventController.getEvent(req.params.eventId);
+        if (currEvent.startDate != null && endDate < currEvent.startDate) {
+          throw new Error("End date is before start date");
+        }
+      }
+    }),
```

Comment:
Same thing I said in an earlier comment regarding the updating event validators for `startDate` and `endDate`. You'll have to modify this to cover all the cases I listed in my earlier comment.